code/modules/modular_computers/ntos/files.dm

```diff
@@ -118,12 +118,14 @@
 
 /datum/file_storage/proc/delete_file(filename)
 
-/datum/file_storage/proc/create_file(newname)
+/datum/file_storage/proc/create_file(newname, var/file_type = /datum/computer_file/data/text)
 	if(check_errors())
 		return FALSE
-	var/datum/computer_file/data/text/F = new()
+	var/datum/computer_file/F = new file_type
 	F.filename = newname
-	F.calculate_size()
+	var/datum/computer_file/data/FD = F
+	if(FD)
```

Comment:
What's this cast/check for?

Reply:
calculate_size() is only present on computer_file/data. I use create_file on some non-/data files, so the interface had to be made more generic and an if added for /data-files.

Reply:
well you shoudl istype here then,  because unless file is null it'll always pass here